command/var_get.go

```diff
@@ -0,0 +1,182 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+)
+
+type VarGetCommand struct {
+	Meta
+	outFmt string
+	tmpl   string
+}
+
+func (c *VarGetCommand) Help() string {
+	helpText := `
+Usage: nomad var get [options] <path>
+
+  The 'var get' command is used to get the contents of an existing variable.
+
+  If ACLs are enabled, this command requires a token with the 'var:read'
+  capability.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault) + `
+
+Read Options:
+
+  -item <item key>
+     Print only the value of the given item. Specifying this option will
+     take precedence over other formatting directives. The result will not
+     have a trailing newline making it ideal for piping to other processes.
+
+  -out ( go-template | hcl | json | none | pretty-json | table )
+     Format to render the variable in. When using "go-template", you must
+     provide the template content with the "-template" option. Defaults
+     to "table" when stdout is a terminal and to "json" when stdout is
+     redirected.
+
+  -template
+     Template to render output with. Required when output is "go-template".
+
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *VarGetCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-out":      complete.PredictSet("table", "hcl", "json", "pretty-json", "go-template"),
+			"-template": complete.PredictAnything,
+		},
+	)
+}
+
+func (c *VarGetCommand) AutocompleteArgs() complete.Predictor {
+	return VariablePathPredictor(c.Meta.Client)
+}
+
+func (c *VarGetCommand) Synopsis() string {
+	return "Read a variable"
+}
+
+func (c *VarGetCommand) Name() string { return "var read" }
+
+func (c *VarGetCommand) Run(args []string) int {
+	var out, item string
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		flags.StringVar(&c.outFmt, "out", "table", "")
+	} else {
+		flags.StringVar(&c.outFmt, "out", "json", "")
+	}
+	flags.StringVar(&c.tmpl, "template", "", "")
+	flags.StringVar(&item, "item", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got one or two arguments
```

Comment:
Fixed the comment. Yeah, not sure if flags can cope with this case well.

Perhaps we could use a [TextVar](https://pkg.go.dev/flag#TextVar) and do extra validation on the `out` flag. However, then we'd still  have the same problem for namespace if you changed the order. Since `-namespace` comes from the common flagset, it might present an adoption challenge for TextVar—a 1.19 addition—into code where backport is a higher probability.

There's [Var](https://pkg.go.dev/flag#Var) flags, which seem like they would give us the validation layer we need IF we want to add the extra scaffolding that they require.